app/src/main/java/com/kickstarter/viewmodels/ChangePasswordViewModel.kt

```diff
@@ -1,165 +1,69 @@
-        private val confirmPassword = PublishSubject.create<String>()
-        private val currentPassword = PublishSubject.create<String>()
-        private val newPassword = PublishSubject.create<String>()
-
-        private val error = BehaviorSubject.create<String>()
-        private val progressBarIsVisible = BehaviorSubject.create<Boolean>()
-        private val success = BehaviorSubject.create<String>()
-
-        val inputs: Inputs = this
-        val outputs: Outputs = this
-
-        private val apolloClient = requireNotNull(this.environment.apolloClientV2())
-        private val analytics = this.environment.analytics()
-
-        private val disposables = CompositeDisposable()
-
-        init {
-
-            val changePassword = Observable.combineLatest(
-                this.currentPassword.startWith(""),
-                this.newPassword.startWith(""),
-                this.confirmPassword.startWith("")
-            ) { current, new, confirm -> ChangePassword(current, new, confirm) }
-
-            val changePasswordNotification = changePassword
-                .compose(takeWhenV2(this.changePasswordClicked))
-                .switchMap { cp -> submit(cp).materialize() }
-                .share()
-
-            changePasswordNotification
-                .compose(errorsV2())
-                .subscribe { error ->
-                    error?.localizedMessage?.let { message ->
-                        this.error.onNext(message)
-                    }
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted.Companion.WhileSubscribed
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.rx2.asFlow
+
+data class UpdatePasswordUIState(
+    val isLoading: Boolean = false,
+    val errorMessage: String? = null,
+    val email: String? = null
+)
+class ChangePasswordViewModel(val environment: Environment) : ViewModel() {
+
+    private val apolloClient = requireNotNull(this.environment.apolloClientV2())
+    private val analytics = requireNotNull(this.environment.analytics())
+
+    private val mutableUIState = MutableStateFlow(UpdatePasswordUIState())
+    val uiState: StateFlow<UpdatePasswordUIState> get() =
+        mutableUIState.asStateFlow()
+            .stateIn(
+                scope = viewModelScope,
+                started = WhileSubscribed(),
+                initialValue = UpdatePasswordUIState(isLoading = true)
+            )
+
+    fun updatePassword(oldPassword: String, newPassword: String) {
+        viewModelScope.launch {
+            // TODO: Avoid using GraphQL generated types such as UpdateUserPasswordMutation.Data, return data model defined within the app.
```

Comment:
Do we want to propose a alternative to this as part of this migration effort?  It would probably mean adding transformers/UiModels to the effort.

Reply:
Agreed, seems a small effort, what do you think @leighdouglas and @ycheng-kickstarter ?

Reply:
I'll take care of this one in a following  PR